src/UI/Buyer/src/app/services/payment-helper/payment-helper.service.ts

```diff
@@ -20,6 +20,25 @@ export class PaymentHelperService {
     return { Items, Meta: payments.Meta }
   }
 
+  async ListAnonPaymentsOnOrder(orderID: string, creditCard?: OrderCloudIntegrationsCreditCardToken): Promise<ListPage<Payment>> {
+    const payments = await Payments.List('Outgoing', orderID);
+    const withDetails = payments.Items.map((payment) => 
+      this.setAnonPaymentDetails(payment, creditCard)
+    )
+    return {
+      Items: withDetails,
+      Meta: payments.Meta
+    }
+  }
+
+  private setAnonPaymentDetails(payment: Payment, creditCard?: OrderCloudIntegrationsCreditCardToken): Payment {
+    const details: BuyerCreditCard = {
+      PartialAccountNumber: creditCard?.AccountNumber?.substr(creditCard?.AccountNumber?.length - 4)
+    }
+    ;(payment as any).Details = details;
```

Comment:
misplaced ; possibly?

Reply:
That's a prettier thing. Since our code style strives to not use semicolons it adds them only when needed. I suppose it does seem more natural if it were on line 37 but functionally equivalent and not something you can really control unless you ignore prettier styling

Reply:
It is weird but typescript is requiring that semi colon. We also follow this pattern in other parts of the app as well. I assume there is a reason for it. Might be worth a post in hs-lint-fire

Reply:
Although I could move it onto the above line. That looks better to me.

Reply:
Actually I can just delete this whole function lol. Update incoming.